src/Tickets/Admin/All_Tickets/List_Table.php

```diff
@@ -189,49 +190,329 @@ public function get_sortable_columns() {
 	}
 
 	/**
-	 * Renders the default column.
+	 * Get the ticket type icon.
 	 *
-	 * @since  TBD
+	 * @since TBD
 	 *
-	 * @param object $item        The current item.
-	 * @param string $column_name The current column name.
+	 * @param Tribe__Tickets__Ticket_Object $item The current item.
+	 *
+	 * @return string HTML for the ticket type icon.
+	 */
+	public function get_ticket_type_icon( $item ) {
+		ob_start();
+		do_action( 'tec_tickets_editor_list_table_title_icon_' . $item->type() );
+		return ob_get_clean();
+	}
+
+	/**
+	 * Get the default column value.
+	 *
+	 * @since TBD
+	 *
+	 * @param Tribe__Tickets__Ticket_Object $item        The current item.
+	 * @param string                        $column_name The column name.
 	 *
 	 * @return string
 	 */
 	public function column_default( $item, $column_name ): string {
-		return $item[ $column_name ];
+		// If the column name is empty or the item does not have the column name, return an empty string.
+		if ( empty( $column_name ) || ! isset( $item->$column_name ) ) {
+			return '';
+		}
+
+		// If value is not a string or a number, return an empty string.
+		if ( ! is_string( $item->$column_name ) && ! is_numeric( $item->$column_name ) ) {
+			return '';
+		}
+
+		return $item->$column_name;
 	}
 
 	/**
-	 * Prepares the list of items for displaying.
+	 * Get the column name value.
 	 *
 	 * @since TBD
+	 *
+	 * @param Tribe__Tickets__Ticket_Object $item The current item.
+	 *
+	 * @return string
 	 */
-	public function prepare_items() {
-		$current_page = $this->get_pagenum();
-		$per_page     = $this->get_items_per_page( $this->per_page_option );
+	public function column_name( $item ): string {
+		$event = $item->get_event();
+		if ( ! $event ) {
+			return '-';
+		}
 
-		$pagination_args = [
-			'total_items' => 1,
-			'per_page'    => $per_page,
+		$edit_post_url  = get_edit_post_link( $event );
+		$edit_post_link = sprintf(
+			'<a href="%s" class="tec-tickets-all-tickets-table-event-link" target="_blank" rel="nofollow noopener">%s</a>',
+			esc_url( $edit_post_url ),
+			esc_html( $item->name )
+		);
+
+		$admin_views = tribe( 'tickets.admin.views' );
+		$context     = [
+			'icon_html'   => $this->get_ticket_type_icon( $item ),
+			'ticket_link' => $edit_post_link,
 		];
 
-		$args = [
-			'page'               => $current_page,
-			'per_page'           => $per_page,
-			'return_total_found' => true,
-			'order'              => 'DESC',
+		return $admin_views->template( 'all-tickets/table-column-name', $context, false );
+	}
+
+	/**
+	 * Get the column ID value.
+	 *
+	 * @since TBD
+	 *
+	 * @param Tribe__Tickets__Ticket_Object $item The current item.
+	 *
+	 * @return string
+	 */
+	public function column_id( $item ): string {
+		return $item->ID;
+	}
```

Comment:
The return here is string or int ? I would update to `return absint( $item->ID );` and the docs to reflect that.

Reply:
This is to print out the ID for the column header, so a string would be ideal. No need to force to an int.

Reply:
Maybe force to a string then its a good idea ? Im just a bit worried if that could ever be an int where we would have an issue with declared return type of the method.